x-pack/plugins/upgrade_assistant/server/lib/apm/index.ts

```diff
@@ -41,15 +41,34 @@ export async function getDeprecatedApmIndices(
   }, []);
 }
 
+// helper to parse out prerelase tags for use with semver
+// https://www.npmjs.com/package/semver#prerelease-tags
+export const getPureVersion = (version: string): string | undefined => {
+  if (!version) {
+    return;
+  }
+
+  const versions = version.split('-');
+  if (!versions) {
+    return;
+  }
+
+  return versions.shift();
+};
+
 export const isLegacyApmIndex = (
   indexName: string,
   apmIndexPatterns: string[] = [],
   mappings: FlatSettings['mappings']
 ) => {
-  const clientVersion = get(mappings, '_meta.version', '0.0.0');
+  const clientVersion = getPureVersion(get(mappings, '_meta.version'));
+
+  if (clientVersion && semver.gte(clientVersion, '7.0.0')) {
+    return false;
+  }
```

Comment:
You should still be able to use SemVer here instead of doing any manual parsing. SemVer instance have a `compareMain` function that ignores the prerelease suffix.

```js
const clientVersion = new SemVer(get(mappings, '_meta.version', '0.0.0'));

if (clientVersion.compareMain('7.0.0') === -1) {
  return false;
}
```